### PR TITLE
Allow for contributions in currencies other than USD

### PR DESCRIPTION
**Pull request by darktrojan**
This isn't ready by any means, but I wanted to get some opinions on it. I'm storing the contribution amount in the original currency, plus the equivalent in USD, since everything currently expects a USD value.

### CODE REVIEW

File: src/olympia/stats/models.py [hunk 1]
```diff
@@ -166,7 +167,17 @@ class Meta:
         db_table = 'stats_contributions'
 
     def __unicode__(self):
-        return u'%s: %s' % (self.addon.name, self.amount)
+        return u'%s: %s %s' % (self.addon.name, self.amount, self.currency)
+
+    def __init__(self, *args, **kw):
+        super(Contribution, self).__init__(*args, **kw)
+        # TODO collect exchange rates from web service and store them somewhere
```
Comment (darktrojan) [inline src/olympia/stats/models.py]: I'm planning on pulling these from openexchangerates.org, on a daily cron job. But how to store them?

**Comment by andymckay** [comment]
Is there a bug for this? I'm needing to think if this is a good idea and how it should be done.

**Comment by darktrojan** [comment]
There's [524556](https://bugzilla.mozilla.org/show_bug.cgi?id=524556), that's about all I could find.

**Comment by andymckay** [comment]
Hmm, that got won't fixed. My first reaction is to say that this is most likely not going to survive the upcoming AMO redesign, but I can't guarantee at this point if it will stay or not. Contributions is cool, but a support burden and people continually file bugs and issues around the paypal integration which is showing its age.

No guarantees on this one, I just don't want you to put in a lot of work that might get deleted in a few months.

File: src/olympia/stats/models.py (hunk 1 above)
Comment (andymckay) [inline src/olympia/stats/models.py]: Why do you need to collect exchange rates at all?

File: src/olympia/migrations/865-add-contributions-usd.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE `stats_contributions` ADD COLUMN `amount_usd` DECIMAL(9, 2);
+UPDATE `stats_contributions` SET `amount_usd` = `amount`;
```
Comment (andymckay) [inline src/olympia/migrations/865-add-contributions-usd.sql]: Why do you need to store the amount in USD, for stats?

Comment (darktrojan) [inline src/olympia/migrations/865-add-contributions-usd.sql]: The stats pages display contributions in USD. Using just the amount would make them worthless (especially given something like the Yen, where $1 = ¥110). The pages could be changed to display amounts for each currency, but this is a lot simpler.

**Comment by darktrojan** [comment]
If I'm reading it right, that bug was wontfixed because everything was being replaced by the marketplace.

File: src/olympia/stats/views.py [hunk 3]
```diff
@@ -454,18 +454,31 @@ def contributions_series(request, addon, group, start, end, format):
...
+          .annotate(count=Count('amount'), total=Sum('amount')))
 
-    # Add `date` and `end` keys for legacy compat.
-    series = sorted(qs, key=lambda x: x['date_created'], reverse=True)
-    for row in series:
-        row['end'] = row['date'] = row.pop('date_created')
+    series = []
+    currencies = set()
+    last = None
+    for row in sorted(qs, key=lambda x: x['date_created'], reverse=True):
```
Comment (EnTeQuAk) [inline src/olympia/stats/views.py]: If `qs` is a django `QuerySet` you might want to do `for row in qs.order_by('-date_created'):` to let the database do the work for you.

Comment (darktrojan) [inline src/olympia/stats/views.py]: Good point. I just borrowed it from the old code without thinking about it too much.

File: static/js/impala/stats/manager.js [hunk 4]
```diff
@@ -8,6 +8,10 @@ z.StatsManager = (function() {
     var STATS_VERSION = '2011-12-12';
     var PRECISION = 2;
 
+    $.ajax('/static/js/currency.json').then(function(c) {
```
Comment (EnTeQuAk) [inline static/js/impala/stats/manager.js]: We usually set `data-static-url` to `<body>` so that you can use that as a base for requesting static files.

**Comment by muffinresearch** [comment]
I think this is unlikely to go ahead since we are trying to minimize what new features we add to the site. 

Since the future of contributions is not certain and needs discussion in the context of any future redesign it doesn't make sense to add more features to it at this time.

I'm going to close this for now but it can stay here and if it makes sense to revisit it in the future we can always re-open it.

**Pull request closed without merging**